### PR TITLE
circle: Suppress Detox build stdout

### LINKED ISSUE #3186: Detox: Suppress build output
Since we build the app in the `Run Fastlane` step, can't we ignore the build output (at least to stdout) when running `yarn detox build --configuration ios.sim.release`? It's basically the same output, but unprettified.

### PR DESCRIPTION
**Pull request by rye**
It's very noisy. Let's turn it down a notch.

Closes #3186.

### CODE REVIEW

**Review by hawkrives — COMMENTED** [review]
Oh, I forgot that there was a separate step. 

Um. 

Given that Circle collapses this row after completion, why do we even need to hide this?

**Comment by rye** [comment]
Given that this output is redundant, why do we need to keep it?

Also, it lags out my browser. It's _very_ noisy.

**Comment by hawkrives** [comment]
Have you tested that it properly prints errors when it errors?

**Comment by hawkrives** [comment]
I'm still not convinced that this won't rob us of debugging output that we might need in the future.

I would prefer to not hide this.

**Comment by rye** [comment]
We should see if we can pipe this to `xcpretty`, then.  I don't like `xcodebuild`'s unnecessary verbosity.

**Comment by rye** [comment]
Interestingly enough, Detox themselves use `xcpretty`. 

https://github.com/wix/Detox/blob/5ef4ae1fa2731b94e95faf0a65920bf9298af0c8/scripts/unit.ios.sh#L3-L4

I have pushed a change that will give that a try.

**Review by hawkrives — DISMISSED** [review]

**Comment by hawkrives** [comment]
I'm nervous about piping test output through xcpretty - it's Jest, IIRC, not Xcode testing, so xcpretty could break it at any time…?

**Comment by rye** [comment]
@hawkrives You will notice that (1) the linked Detox unit `test` step pipes to `xcpretty`, and (2) build [35185](https://circleci.com/gh/StoDevX/AAO-React-Native/35185) includes piping non-`xcodebuild` output to `xcpretty`, and does not fail.

**Comment by rye** [comment]
As such, I'm pretty sure xcpretty is just doing pattern matching. If it doesn't find any patterns, they don't get formatted.

**Comment by hawkrives** [comment]
Yes, but I'm asking if we can rely on that not breaking in the future.

**Comment by hawkrives** [comment]
It definitely hides some of the output lines

**Comment by hawkrives** [comment]
I'd need to at the _least_ see a test failure piped through Detox to make sure it doesn't hide the actual fails, because that'd be bad.

File: .circleci/config.yml [hunk 1]
```diff
@@ -338,8 +338,8 @@ jobs:
           command: bundle exec fastlane ios ci-run | tee ./logs/build
           environment:
             GIT_COMMIT_DESC: $(git log --format=oneline -n 1 $CIRCLE_SHA1)
-      - run: yarn detox build --configuration ios.sim.release
-      - run: yarn detox test --configuration ios.sim.release --cleanup
+      - run: yarn detox build --configuration ios.sim.release | xcpretty
+      - run: yarn detox test --configuration ios.sim.release --cleanup | xcpretty
```
Comment (hawkrives) [inline .circleci/config.yml]: Let's just not pipe test output through `xcpretty`

```suggestion
      - run: yarn detox test --configuration ios.sim.release --cleanup
```

**Comment by rye** [comment]
I will make a branch off of this branch, invert at least one of the assertions in our tests, and watch that fail. We'll see if it actually slurps the output. (I'm 99% sure it'll still work.)

If you actually read [the business logic in the xcpretty source](https://github.com/supermarin/xcpretty/blob/4b90228e80dd74dbdb209553668178023447ff85/bin/xcpretty#L83-L86), xcpretty is simply reading from STDIN and writing to STDOUT, and the [printer](https://github.com/supermarin/xcpretty/blob/4b90228e80dd74dbdb209553668178023447ff85/lib/xcpretty/printer.rb#L18-L24) is simply forwarding (in our case to the [simple](https://github.com/supermarin/xcpretty/blob/4b90228e80dd74dbdb209553668178023447ff85/lib/xcpretty/formatters/simple.rb) formatter) the input; if Jest manages to output something that triggers an EMPTY handler, then that's really special and we can handle that.

**Review by hawkrives — DISMISSED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (rye) [reply .circleci/config.yml]: Why, though? I still don't understand how xcpretty, which is only slurping stuff that looks like `xcodebuild` output, would possibly slurp jest output.  Expect the Proof of Concept that it actually does report failures shortly, but I see no reason to make an exception and keep the noise in the test output, especially if that's wholly unnecessary.

**Comment by hawkrives** [comment]
@rye okay, but what does piping our Jest test output to xcpretty actually get us? even hypothetically.

I see only potential downsides, and no reason to pipe it in.

File: .circleci/config.yml (hunk 1 above)
Comment (hawkrives) [reply .circleci/config.yml]: I see no reason to even pipe it through. By that argument, why not pipe literally every command through xcpretty? 

I just don't see any reason to pipe this through.

**Comment by rye** [comment]
> @rye okay, but what does piping our Jest test output to xcpretty actually get us? even hypothetically.

The output of detox is not only Jest.  There's also xcodebuild output, is there not?

- - -

Okay, there's not.

image

becomes

image

- - -

Fine, I'll remove it.  But not because it might be eating output—only because there isn't `xcodebuild` output. Also, the colors go missing when it goes through `xcpretty`.

**Comment by hawkrives** [comment]
> Fine, I'll remove it. But not because it might be eating output—only because there isn't xcodebuild output. Also, the colors go missing when it goes through xcpretty.

I mean, that was kinda my argument – there's no point to sending the test output through xcpretty because there's nothing but `jest` output.

And, I will note that it _is_ eating output, currently – there's no more `yarn` ending line, for instance.

**Review by hawkrives — APPROVED** [review]

**Comment by rye** [comment]
FWIW, here is a [deliberate failure](https://circleci.com/gh/StoDevX/AAO-React-Native/35197) which demonstrates that failure context is not messed with.

But whatever.

**Pull request merged**